apps/language_server/lib/language_server/experimental/code_mod/diff.ex

```diff
@@ -70,7 +70,15 @@ defmodule ElixirLS.LanguageServer.Experimental.CodeMod.Diff do
 
   defp apply_diff(:ins, {line, code_unit} = position, change, {current_line, prev_lines}) do
     current_line = [edit(change, line, code_unit, line, code_unit) | current_line]
-    advance(change, position, {current_line, prev_lines})
+    advance_ins(change, position, {current_line, prev_lines})
+  end
+
+  defp advance_ins(<<>>, position, edits) do
+    {position, edits}
+  end
+
+  defp advance_ins(<<_c::utf8, rest::binary>>, position, edits) do
+    advance_ins(rest, position, edits)
```

Comment:
Why introduce the new function and iterate over the string if all it does is a no-op? Perhaps a comment here about what it's doing and returning `{position, {current_line, prev_lines}}` is all that's needed.